crates/scroll/evm/src/execute.rs

```diff
@@ -263,6 +268,21 @@ where
     }
 }
 
+/// Helper type with backwards compatible methods to obtain Scroll executor
+/// providers.
+#[derive(Debug)]
+pub struct ScrollExecutorProvider;
+
+impl ScrollExecutorProvider {
+    /// Creates a new default scroll executor provider.
+    pub fn scroll(
+        // TODO (scroll): replace with `ScrollChainSpec`.
+        chain_spec: Arc<ChainSpec>,
+    ) -> BasicBlockExecutorProvider<ScrollExecutionStrategyFactory> {
+        BasicBlockExecutorProvider::new(ScrollExecutionStrategyFactory::new(chain_spec))
+    }
```

Comment:
question: Could we benefit from having `scroll_mainnet()` and `scroll_sepolia()` methods on this as well? Not sure if they would be used. What's the benefit of using this type? It looks like we don't use it in `ScrollExecutorBuilder`.

Reply:
This is used to launch the [stage command](https://github.com/paradigmxyz/reth/blob/735eb4b97cf3937886a2972647e0afb90c916a71/crates/cli/commands/src/stage/mod.rs#L47-L48). We could provide just a closure but this `Provider` approach is what op and ethereum use